### ISSUE TITLE
Web audit for qiskit.org/documentation

### ISSUE THREAD

**Issue by y4izus** [Qiskit/qiskit-metapackage#1418]
With [Brightedge](https://brightedge.com) we can do site audits that helps us crawling and resolving technical errors on the website to improve performance. By doing periodic audits we can see the evolution of our web performance, the content and page structure score and the crawlability. This is useful information because:

- Optimizing web performance is critical to driving traffic, generating more leads and improving conversion rates
- A good site structure means a great user experience. It is crucial that all the page tags on the site are structured correctly, with appropriate page titles, headings, meta descriptions, etc. An accurate site structure can reduce bounce rate and increase dwell time, both of which may help boost our search performance
- Every bad or broken internal link on the website reduce the crawlability rating of the web which may impact the search performance

To do the web audit go to Brightedge -> Action -> Site Audit -> select project and click on `Start Crawl`

**Comment by y4izus** [comment]
I did the first audit on December and realized that several stable versions for qiskit documentation were broken ( #1389 ). After fixing that issue, the errors on the audit changed from:
<img width="155" alt="Captura de pantalla 2022-02-15 a las 17 18 06" src="https://user-images.githubusercontent.com/17231966/154103252-a107a3b4-2770-4d55-9788-d877c9b10a4a.png">

to

<img width="161" alt="Captura de pantalla 2022-02-15 a las 17 18 40" src="https://user-images.githubusercontent.com/17231966/154103356-1b88b9af-8097-4b9c-97e8-826848c5995e.png">

Just for 4xx Erros and Broken internal links, we reduced the number of errors by 80870:

<img width="1345" alt="Captura de pantalla 2022-02-15 a las 17 20 41" src="https://user-images.githubusercontent.com/17231966/154103721-85851974-0876-4b1b-8e59-4ad09f0e5f71.png">

[Link](https://app8.brightedge.com/ui/platform/contentiq/details?errorType=0&pageType=-1&groupBy=errors&crawlConfigurationId=6552&diffCrawlId=86619&crawlId=87061) to the audit done after the issue was solved by @mtreinish and @jakelishman . Thank you both!

**Comment by y4izus** [comment]
A [new audit](https://app8.brightedge.com/ui/platform/contentiq?crawlId=87061) has been done on Feb. 13, 2022. Mainly issues found:

- Severe errors
  - Duplicate page title
  - 4xx Errors
  - Duplicate page content

- Moderate errors:
  - Missing or empty meta description tag
  - Page title too short or too long
  - Broken internal links

- Minor errors:
  - Too many outbound internal links
  - 3xx redirects

[TODO] Identify errors on the code and create issues

**Comment by jakelishman** [comment]
I can't access the audits - I don't seem to have any login for BrightEdge - so I might be a little off the mark here.  For "duplicate page content", is there a risk that it's going to be complaining about multiple versions of documentation?

**Comment by y4izus** [comment]
I'm not sure if that is the reason of the duplicated content. Mostly of the duplicated pages are when comparing versions 0.25 and 0.26 but with words in the URL like `stubs` or `_modules`. Could be that we are pushing content that shouldn't be on production?

Some examples:
![Captura de pantalla 2022-02-16 a las 16 48 04](https://user-images.githubusercontent.com/17231966/154302508-41693ba2-5600-4a3d-bc41-07aef3611bd7.png)

**Comment by jakelishman** [comment]
Looking at your picture, that's exactly what I meant - it's complaining when the documentation didn't change between versions.  `stubs` will contain the majority of the actual documentation as a side-effect of how its built by `autosummary`, though I would not at all be opposed to reducing our reliance on `autosummary`.

**Comment by y4izus** [comment]
> Looking at your picture, that's exactly what I meant - it's complaining when the documentation didn't change between versions. `stubs` will contain the majority of the actual documentation as a side-effect of how its built by `autosummary`, though I would not at all be opposed to reducing our reliance on `autosummary`.

Would you mind creating an issue to investigate that proposal? It's not a priority right now, but I think we should take a look at it and I don't want to have it lost on this issue

The audit is done so I'm going to close this issue. Within the errors found, we should prioritize #1438 . The rest of the errors are to improve SEO and could be done in the future

**Comment by jakelishman** [comment]
Those ones I flagged are (to some degree) false-positive results, so I'm not certain if it's an issue on Qiskit.  Those pages are the same, but they're _meant_ to be the same - `qiskit.org/documentation/<version>/` is an archive of the documentation build for one particular version of the package.  If a particular class or function doesn't change between versions, its documentation will likely be the same, and it'll generate the same content.  I've got a rough hope that one day we might be able to put a banner on old pages saying "this is old documentation, click here to go to this page in the newest documentation", but achieving that (correctly) is quite tricky.

(Removing `autosummary` wouldn't change the end result that qiskit.org needs to have many near-duplicate pages on it, it'd just change how we went about generating them.)
